planet/src/main/webapp/places.js

```diff
@@ -22,48 +24,61 @@ function initMap() {
         center: TORONTO_COORDINATES,
         zoom: 8
     });
-    let infoWindow = new google.maps.InfoWindow;
-        // Gets users current position
-        if (navigator.geolocation) {
-            navigator.geolocation.getCurrentPosition(function(position) {
-            let pos = {
-              lat: position.coords.latitude,
-              lng: position.coords.longitude
+
+    // Gets users current location
+    if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function(position) {
+            pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude
             };
-            infoWindow.setPosition(pos);
-            infoWindow.setContent('You are here.');
-            infoWindow.open(map);
+            // Create draggable marker at current location
+            let searchMarker = new google.maps.Marker({
+                position: pos,
+                map: map,
+                animation: google.maps.Animation.DROP,
+                draggable: true
+            });
             map.setCenter(pos);
         }, function() {
-            // User did not allow location
-            handleLocationError(true, infoWindow, map.getCenter());
-          });
-        } else {
-            // Browser doesn't support Geolocation
-            handleLocationError(false, infoWindow, map.getCenter());
-        }
-    updateSearch();
-}
+            let searchMarker = new google.maps.Marker({
+                position: TORONTO_COORDINATES,
+                map: map,
+                animation: google.maps.Animation.DROP,
+                draggable: true
+            });
+            map.setCenter(TORONTO_COORDINATES);
+            
+            });
+    } else {
+        alert('Error: Your browser doesn\'t support geolocation.'); 
+    }
 
-/** Create new markers and display new search */
-function updateSearch() {    
     let input = document.getElementById('pac-input');
     let searchBox = new google.maps.places.SearchBox(input);
 
     // Set position of the search bar onto the map
     map.controls[google.maps.ControlPosition.TOP_LEFT].push(input);
+    
+    updateSearch();
+}
+
+/** Create new markers and display new search */
+function updateSearch() {    
+    input = document.getElementById('pac-input');
+    searchBox = new google.maps.places.SearchBox(input);
```

Comment:
Did you forget to declare these two?

Reply:
They are declared in the previous function initMap() - this is to set the position of the search bar

Reply:
They're not in the same scope. As written now, these are globally scoped. If you need them to be global variables, declare them at the global level. Otherwise, drop the let statements on 56 and 57 and use 'let' here.

https://www.w3schools.com/js/js_scope.asp